src/components/EditorDisplay.tsx

```diff
@@ -175,9 +175,10 @@ export function EditorDisplay(props: EditorDisplayProps) {
     flushSync(() => {
       setLocalIndex(localIndex + 1);
 
+      // TODO: this shouldn't be necessary. Moving to redux may make this less painful.
       const updated = datastore.update(currentItem!, value || "");
       if (updated && updated.pathname !== location.pathname) {
-        history.replace(updated.pathname);
+        navigate({ to: updated.pathname, replace: true });
```

Comment:
There's lots of this kind of change, because navigate behaves differently than the old `history` stuff did. I think they made that API change mostly to make `navigate` compatible with suspense and dataloading logic, which `history`'s API couldn't really.